xline/src/storage/auth_store/backend.rs

```diff
@@ -253,30 +255,32 @@ impl AuthStoreBackend {
     }
 
     /// get `KeyValue` in `AuthStore`
-    fn get(&self, key: &[u8]) -> Option<KeyValue> {
+    fn get(&self, key: &[u8]) -> Result<Option<KeyValue>, ExecuteError> {
         let revisions = self.index.get(key, &[], 0);
         assert!(revisions.len() <= 1);
-        self.db.get_values(&revisions).pop()
+        self.db.get_values(&revisions).map(|mut kv| kv.pop())
     }
 
     /// get user by username
     pub(super) fn get_user(&self, username: &str) -> Result<User, ExecuteError> {
         let key = [USER_PREFIX, username.as_bytes()].concat();
-        match self.get(&key) {
-            Some(kv) => Ok(User::decode(kv.value.as_slice()).unwrap_or_else(|e| {
-                panic!(
-                    "Failed to decode user from kv value, error: {:?}, kv: {:?}",
-                    e, kv
-                )
-            })),
-            None => Err(ExecuteError::InvalidCommand("user not found".to_owned())),
-        }
+        self.get(&key)?.map_or_else(
```

Comment:
Can we keep the code style consistent with that in `get_role`?

Reply:
fixed